File: src/main/resources/static/js/app/stockservice.js
'use strict'
angular.module('stock.services', []).factory('stockservice', ["$http", "CONSTANTS", function($http, CONSTANTS) {
    var service = {};
    service.getAllStocks = function() {
        return $http.get(CONSTANTS.getAllStocks);
    }
    service.getStockById = function(stockId) {
        var url = CONSTANTS.getStockById + stockId;
        return $http.get(url);
    }
    service.saveStock = function(stockDto) {
        return $http.post(CONSTANTS.saveStock, stockDto);
    }
    service.updateStock = function(stockDto) {
    	var url = CONSTANTS.updateStock + stockDto.stockId;
        return $http.put(url, stockDto);
    }
    service.deleteAllStocks = function() {
        return $http.delete(CONSTANTS.deleteAllStocks);
    }
    service.deleteStock = function(stockDto) {
    	var url = CONSTANTS.deleteStockById + stockDto.stockId;
        return $http.delete(url, stockDto);
    }
    return service;
}]);

File: src/main/java/com/payconiq/stocks/service/StockService.java
package com.payconiq.stocks.service;

import java.util.List;

import com.payconiq.stocks.entity.Stock;
import com.payconiq.stocks.model.StockDto;

public interface StockService {

  /**
   * Get all stocks.
   * 
   * @return List list of all Stocks present in database.
   */
  List<StockDto> getAllStocks();

  /**
   * Get a stock by Id.
   * 
   * @param stockId id representing which stock should be fetched.
   * @return StockDto a particular stock based on the stockId present.
   */
  StockDto getStockById(long stockId);

  /**
   * Save stocks.
   * 
   * @param stockDto a particular stock to be saved in database.
   * @return Stock what has been saved.
   */
  Stock saveStock(StockDto stock);

  /**
   * Update a particular stock.
   * 
   * @param stockDto dto sent which has to be updated.
   * @param stockId stockId passed which has to be updated.
   * @return Stock which has been updated.
   */
  Stock updateStock(StockDto stockDto, long stockId);

  /**
   * Delete all stocks.
   */
  void deleteAllStocks();

  /**
   * delete controller to delete a particular stock.
   * 
   * @param stockId delete by stockId.
   */
  void deleteStockById(long stockId);

}
